### PR TITLE
dynamically render image gallery on style toggle

### PR DESCRIPTION
**Pull request by the-trash-panda**
- image gallery change on style thumbnail click
- cart component quantity/size change on style thumbnail click

### CODE REVIEW

File: client/src/components/shared/calcStarImg.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ const calcStarImg = (numStars) => {
   let starImg = [];
   let wholeStars = Math.floor(numStars);
   let fractionStar = numStars - Math.floor(numStars);
-  let emptyStars = 5 - wholeStars;
+  let emptyStars = 5 - Math.ceil(numStars);
```
Comment (awhlam) [inline client/src/components/shared/calcStarImg.jsx:7]: Thanks for catching and fixing this!

File: client/src/components/overview/ProductInfo.jsx [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+import calcStarImg from '../shared/calcStarImg.jsx';
 
-const ProductInfo = function ({ product, reviewsMeta }) {
+const ProductInfo = function ({ product, reviewMetaData }) {
   let sum = 0;
   let reviews = 0;
-  for (const key in reviewsMeta.ratings) {
-    sum += Number(reviewsMeta.ratings[key]) * Number(key);
-    reviews += Number(reviewsMeta.ratings[key]);
+  for (const key in reviewMetaData.ratings) {
```
Comment (awhlam) [inline client/src/components/overview/ProductInfo.jsx:7]: You could use the function I wrote in the shared folder to get the average and total number of reviews if you want:

https://github.com/Audacious-Alder/FEC/blob/main/client/src/components/shared/calcAvgTotalReviews.js

**Review by awhlam — APPROVED** [review]
Looks good! I left one suggestion about getting the average and total number of reviews.

You can make a commit for that suggestion, or feel free to merge this PR now if you want to consider it for later.

File: client/src/components/overview/ProductInfo.jsx (hunk 2 above)
Comment (the-trash-panda) [reply client/src/components/overview/ProductInfo.jsx]: Thanks for the suggestion, I just fixed it!

**Comment by awhlam** [comment]
@the-trash-panda I think this PR is causing an issue where it's trying to render Overview data before the `axios.get` request for the Overview data is complete (so it ends up trying to render undefined).

The way I handled it for reviews is by only rendering if the data is not undefined: https://github.com/Audacious-Alder/FEC/blob/main/client/src/components/App.jsx#L41

I made a PR today to refactor the data fetching that should solve this for both of us: PR #30

**Pull request merged**